docs/source/cli/model-training-inference/distributed/sagemaker.rst

```diff
@@ -370,187 +369,52 @@ Using ``Processor`` arguments the above example would become:
-* **--graph-name**: Name of the input graph. The graph name must adhere to the `Python identifier naming rules<https://docs.python.org/3/reference/lexical_analysis.html#identifiers>`_ with the exception that hyphens (``-``) are permitted and the name can start with numbers.
-* **--yaml-s3**: S3 location of yaml file for training and inference.
-* **--custom-script**: Custom training script provided by customers to run customer training logic. This should be a path to the Python script within the Docker image.
-* **--output-emb-s3**: S3 location to store GraphStorm generated node embeddings. This is an inference only argument.
-* **--output-prediction-s3**: S3 location to store prediction results. This is an inference only argument.
-
-Run GraphStorm on Docker Compose for Training
-```````````````````````````````````````````````
-First, use the following command to generate a Compose YAML file for the Link Prediction training on OGB-MAG graph.
-
-.. code-block:: bash
-
-    python3 generate_sagemaker_docker_compose.py \
-            --aws-access-key <<AWS_ACCESS_KEY>> \
-            --aws-secret-access-key <AWS_SECRET_ACCESS_KEY> \
-            --aws-session-token <AWS_SESSION_TOKEN> \
-            --num-instances 3 \
-            --image <GRAPHSTORM_DOCKER_IMAGE> \
-            --graph-data-s3 s3://<PATH_TO_DATA>/ogbn_mag_lp_3p \
-            --yaml-s3 s3://<PATH_TO_TRAINING_CONFIG>/map_lp.yaml \
-            --model-artifact-s3 s3://<PATH_TO_SAVE_TRAINED_MODEL> \
-            --graph-name ogbn-mag \
-            --task-type link_prediction \
-            --num-layers 1 \
-            --fanout 10 \
-            --hidden-size 128 \
-            --backend gloo \
-            --batch-size 128
-
-The above command will create a Docker Compose file named ``docker-compose-<task-type>-<num-instances>-train.yaml``, which we can then use to launch the job.
-
-As our Docker Compose will use a Docker network, named ``gsf-network``, for inter-container communications, users need to run the following command to create the network before luanch Docker Compose.
-
-.. code-block:: bash
-
-    docker network create "gsf-network"
+.. code:: bash
 
-Then, use the following command to run the Link Prediction training on OGB-MAG graph.
+    pip install 'sagemaker[local]' --upgrade
 
-.. code-block:: bash
+second, ensure your local mode configuration includes a high shared memory size for the
+launched local containers:
 
-    docker compose -f docker-compose-link_prediction-3-train.yaml up
+.. code:: yaml
 
-Running the above command will launch 3 instances of the image, configured with the command and env vars that emulate a SageMaker execution environment and run the ``sagemaker_train.py`` script.
+    local:
+        local_code: true # Using everything locally
+        region_name: "us-west-2" # Name of the region
+        container_config:
+            shm_size: "32G" # Set this according to your available memory
```

Comment:
Is it required or will it have default? We can suggest the users to set as their memory size - 256MB

Reply:
There's no default, I added the recommendation to set to 90% of main memory size. I'll see if I can add an early warning so users are not confused by bus errors.